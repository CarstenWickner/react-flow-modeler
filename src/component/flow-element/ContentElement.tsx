import * as React from "react";

export const ContentElement: React.FunctionComponent<{
    children: React.ReactChild;
}> = ({ children }) => (
    <>
        <div className="arrow" />
        <div className="flow-element content-element">{children}</div>
    </>
);
