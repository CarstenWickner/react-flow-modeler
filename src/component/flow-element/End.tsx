import * as React from "react";

export const End: React.FunctionComponent = () => (
    <>
        <div className="arrow" />
        <div className="flow-element end-element" />
    </>
);
