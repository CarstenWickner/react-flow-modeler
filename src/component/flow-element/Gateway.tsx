import * as React from "react";

import { HorizontalStroke } from "./HorizontalStroke";

export const Gateway: React.FunctionComponent<{ type: "converging" } | { type: "diverging"; children?: React.ReactChild }> = ({ type, children }) => (
    <>
        <div className="arrow" />
        <div className={`flow-element gateway-element ${type}`} />
        {type === "diverging" && <HorizontalStroke>{children}</HorizontalStroke>}
    </>
);
