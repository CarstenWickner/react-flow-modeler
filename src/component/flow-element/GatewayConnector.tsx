import * as React from "react";

import { HorizontalStroke } from "./HorizontalStroke";

export const GatewayConnector: React.FunctionComponent<{
    incomingConnection: "first" | "middle" | "last";
    children?: React.ReactChild;
}> = ({ incomingConnection, children }) => <HorizontalStroke incomingConnection={incomingConnection}>{children}</HorizontalStroke>;
