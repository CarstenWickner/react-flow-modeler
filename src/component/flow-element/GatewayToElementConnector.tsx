import * as React from "react";

import { HorizontalStroke } from "./HorizontalStroke";

export const GatewayToElementConnector: React.FunctionComponent<{
    connectionType: "first" | "middle" | "last";
    children?: React.ReactChild;
}> = ({ connectionType, children }) => <HorizontalStroke incomingConnection={connectionType}>{children}</HorizontalStroke>;
