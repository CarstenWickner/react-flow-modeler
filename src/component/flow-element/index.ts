export { Start } from "./Start";
export { ContentElement } from "./ContentElement";
export { Gateway } from "./Gateway";
export { GatewayConnector } from "./GatewayConnector";
export { StrokeExtension } from "./StrokeExtension";
export { End } from "./End";
