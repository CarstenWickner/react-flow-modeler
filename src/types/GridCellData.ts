export interface GridCellCoordinates {
    colStartIndex: number;
    colEndIndex?: number;
    rowStartIndex: number;
    rowEndIndex?: number;
}

export interface SimpleElement extends GridCellCoordinates {
    type: "start" | "end" | "stroke-extension";
}

/*
 * contextual information concerning a flow element
 */
export interface ContentCellData extends GridCellCoordinates {
    type: "content";
    elementId: string;
    data?: { [key: string]: unknown };
}
export interface GatewayDivergingCellData extends GridCellCoordinates {
    type: "gateway-diverging";
    gatewayId: string;
    data?: { [key: string]: unknown };
}
export interface GatewayToElementConnectorCellData extends GridCellCoordinates {
    type: "gateway-to-element";
    gatewayId: string;
    elementId?: string;
    data?: { [key: string]: unknown };
    connectionType: "first" | "middle" | "last";
}
export interface ElementToGatewayConnectorCellData extends GridCellCoordinates {
    type: "element-to-gateway";
    elementId: string;
    connectionType: "first" | "middle" | "last";
}
export interface GatewayConvergingCellData extends GridCellCoordinates {
    type: "gateway-converging";
}

export type GridCellData =
    | SimpleElement
    | ContentCellData
    | GatewayDivergingCellData
    | GatewayToElementConnectorCellData
    | GatewayConvergingCellData
    | ElementToGatewayConnectorCellData;
