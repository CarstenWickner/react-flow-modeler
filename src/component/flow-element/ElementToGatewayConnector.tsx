import * as React from "react";

import { HorizontalStroke } from "./HorizontalStroke";
import { ConnectionType } from "../../types/GridCellData";

export const ElementToGatewayConnector: React.FunctionComponent<{
    connectionType: ConnectionType;
}> = ({ connectionType }) => <HorizontalStroke outgoingConnection={connectionType} />;
