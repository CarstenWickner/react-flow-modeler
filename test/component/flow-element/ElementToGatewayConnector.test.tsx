import * as React from "react";
import { shallow } from "enzyme";

import { ElementToGatewayConnector } from "../../../src/component/flow-element/ElementToGatewayConnector";
import { ConnectionType } from "../../../src/types/GridCellData";

describe("renders correctly", () => {
    it("with minimal/default props", () => {
        const component = shallow(<ElementToGatewayConnector connectionType={ConnectionType.First} />);
        expect(component).toMatchSnapshot();
    });
});
