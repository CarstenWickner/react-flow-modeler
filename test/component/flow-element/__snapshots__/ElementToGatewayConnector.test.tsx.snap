// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with minimal/default props 1`] = `
<HorizontalStroke
  incomingConnection={null}
  outgoingConnection={1}
/>
`;
